src/copy-to-clipboard/internal.tsx

```diff
@@ -29,20 +29,25 @@ export default function InternalCopyToClipboard({
 
   const baseProps = getBaseProps(restProps);
   const onClick = () => {
-    if (navigator.clipboard) {
-      setStatus('pending');
-      setStatusText('');
-      navigator.clipboard
-        .writeText(textToCopy)
-        .then(() => {
-          setStatus('success');
-          setStatusText(copySuccessText);
-        })
-        .catch(() => {
-          setStatus('error');
-          setStatusText(copyErrorText);
-        });
+    if (!navigator.clipboard) {
+      // The clipboard API is not available in insecure contexts.
+      setStatus('error');
+      setStatusText(copyErrorText);
```

Comment:
It seems like we have a UX problem of showing the same error text for all cases. This does not allow users to understand how to fix the copying (clicking the button again obviously won't help here)

Reply:
Yes, this is a general problem in this component. We also don't expose the error reasonto the developers (whether it's the missing API or a thrown error) . Maybe we should add `console.error` commands in this component (although we're not doing that in any other components, as far as I know).

Reply:
We can merge this as is and raise this question with our team (open a ticket if this use-case is accepted)